### ISSUE TITLE
Redirect on submission not working

### ISSUE THREAD

**Issue by gah242s** [progressivetech/net.ourpowerbase.remoteform#39]
I didn't see this in any of the issues, and maybe I've got a site related problem.  However, the profile is set to redirect to a URL after submission.  However, after doing so via the RemoteForm, the redirect does not happen, but the green success bar goes across and everything else is happy.  Thoughts?  Places to look?

**Comment by jmcclelland** [comment]
I haven't given this much thought - I but I think I would consider this by design. There are some aspects of a profile that should definitely get triggered by remote form (like email notification and addition to a group). But I would (initially at least) consider redirect as something that should be explicitly skipped - since one big reason for using Remoteform is to control the user experience.

Just my first reactions! Ultimately I don't have a very strong opinion.

**Comment by gah242s** [comment]
Hey Jamie, thanks for helping think it through.  The idea is a redirect after the submission for things like a landing page.  So, I've got a landing page where they fill out the info, then get redirected to the page containing the information or a follow up thank you for swag registration, etc.  Not a paywall, but more of a light validation form idea.  The landing page is on a remote server, and configuring RemoteForms is multitudes simpler than adjusting the HTML for profiles, especially when making changes to the form later.  Perhaps we have different definitions of what it means to control the user experience, but the idea here is to control what happens to the user when they click submit.

**Comment by gah242s** [comment]
Just a thought, but what if this was a commented out option on the code, much like the payment test section?  That way we wouldn't change existing functionality, and would be a simple, easily seen option that also doesn't change the look and feel of the form.

**Comment by jmcclelland** [comment]
I see what you mean. I wonder if anyone might have a profile that is available on their CiviCRM site and they want users redirected, but they also make it available as a remote form and they _don't_ want users redirected on the remote form? Seems unlikely I guess.

I think we could pull in whatever the configured redirect URL is and redirect the user after the form is submitted via javascript. If we wanted to be as flexible as possible, we could also provide a configuration option to NOT redirect the user, but that might just be over kill. If you don't want to be redirected, you can simply edit the profile and remove the link.

**Comment by gah242s** [comment]
Another way to do this might be to use the cfg.api calls?  I successfully inserted a live URL into the cfg.successMsg by using a single quote, 'link', for the href, since it was encapsulated in a double quote, "text", situation.  Didn't test all the browsers, but Safari liked it and that was good enough for the quick test.  Anyways, could that be reset to a full page bounce instead of just text?

**Comment by jmcclelland** [comment]
Ah - so when the profile is filled out, the user is presented with a clickable link to take them to the page? 

I don't think it would be a good idea to program remoteform to redirect the user's browser to whatever is set for cfg.successMsg - because if someone set that to a regular text message instead of an URL, it would result in an error.

**Comment by hesco** [comment]
> I think we could pull in whatever the configured redirect URL is and redirect the user after the form is submitted via javascript. If we wanted to be as flexible as possible, we could also provide a configuration option to NOT redirect the user, but that might just be over kill. If you don't want to be redirected, you can simply edit the profile and remove the link.

I like this plan for using the redirect configured on the profile, and like the configuration option to disable the redirect.  I would urge support for:

cfg.successRedirectURL
cfg.successRedirectDisable

**Comment by shaneonabike** [comment]
I agree with having new variables that allow for redirection. It's a lot more explicit. Is there an easy way to get this working in RemoteForm? Can I help to make that happen.

**Comment by jmcclelland** [comment]
Yes - I support the idea of using new variables for this. I don't have the time to add this myself, but would merge a PR if someone is able to create one.
